src/file.ts

```diff
@@ -874,6 +876,9 @@ class File extends ServiceObject<File> {
       pathPrefix: '/acl',
     });
 
+    this.crc32cGenerator =
+      options.crc32cGenerator || this.bucket.crc32cGenerator;
```

Comment:
Is there a reason to look to the bucket if one is not passed as an option? Should there be a default within the File class?

Reply:
It appears all `File`s must have a bucket (looking at the construction) and all buckets have a generator (whether custom or the default) - this way customers can set things at a storage, bucket, or per-object level